### PR TITLE
Update README files

### PR DESCRIPTION
**Pull request by yuwzho**
* add "what you need" to instruct reader to install necessary tools
* fix wrong mentions "BLE sample" in "simulate device sample"
* add how to create Azure Function application

### CODE REVIEW

**Comment by yuwzho** [comment]
@yaweiw

File: Lesson3/README.md [hunk 1]
```diff
@@ -12,9 +12,13 @@ See [Lesson 3: Configure and run the BLE sample application to receive data from
 ```
 
 ## Running this sample
+
 Please follow the [Lesson 3: Configure and run the BLE sample application to receive data from BLE SensorTag](#) for detailed walkthough of the steps below.
 
 ### Deploy and run BLE sample
+
+> If you don't have a physical SensorTag device, please jump to [Deploy and run simulate device cloud upload sample](#run-simudev)
```
Comment (yaweiw) [inline Lesson3/README.md]: If you don't have TI SensorTag on hands, you can go to [Deploy and run simulate device cloud upload sample](#run-simudev)

File: Lesson3/README.md [hunk 2]
```diff
@@ -60,12 +64,18 @@ Please follow the [Lesson 3: Configure and run the BLE sample application to rec
...
+## <a id="run-simudev"></a>Deploy and run simulate device cloud upload sample
+1. Use the following command to modify your configuration, change the `"has_sensortag"` value from `true` to `false`.
+
+   ```bash
+   code ../config.json
+   ```
 2. **`npm install`** - Install the npm packages
-3. **`gulp init`** - Intialize configuration files in user's profile folder `%USERPROFILE%\.iot-hub-getting-started`
+3. > *If you have run the script with `has_sensortag: true` flag before, you can run `gulp clean` to clean your personal config, this command will clean your edited config files in your profile folder created by `gulp init`, and tools installed on NUC by `gulp install-tools` command*
+
```
Comment (yaweiw) [inline Lesson3/README.md:75]: *If you have run the script with `has_sensortag: true` before, you can run `gulp clean` to clean your environment. This command removes config files generated by `gulp init` in your profile folder as well as tools installed by `gulp install-tools` on the NUC*

File: Lesson4/README.md [hunk 3]
```diff
@@ -19,11 +19,22 @@ See [Lesson 4: Save messages to the cloud and read them from the cloud](#) for m
...
+   az resource group deployment create --template-file arm-template.json --parameters @arm-template-param.json -g {resource group name}
+   ```
+
+
 ### Deploy and run
 1. **`npm install`** - Install the npm packages
 2. **`gulp init`** - Intialize config files in user's profile folder `%USERPROFILE%\.iot-hub-getting-started`
     - `config-gateway.json`: Configuration for connecting to gateway via SSH.
-    - `config-sensortag.json`: Configuration for the SensorTag and BLE sample application in the Gateway SDK.
+    - `config-sensortag.json`: Configuration for sample application in the Gateway SDK and your physical SensorTag(if you have a one).
```
Comment (yaweiw) [inline Lesson4/README.md]: and your TI SensorTag (if you have one).

File: README.md [hunk 4]
```diff
@@ -5,20 +5,25 @@ This repo contains the sample codes to help you run the azure-iot-gateway-sdk BL
...
+## The following scripts will help you
 * Device discovery for sensortag devices
 * Test the connectivity of sensortag devices
 * Run the BLE sample application on Intel NUC
 * Receive messages from your IoT hub
 * Receive messages from your Azure Table storage
 
+## What you need
+* Set up your Intel NUC as an IoT gateway (Lesson [Set up Intel NUC as an IoT gateway](#))
+* Install Nessary tools on your host machine (Lesson [Get the tools](#))
```
Comment (yaweiw) [inline README.md]: necessary

**Review by yaweiw — APPROVED** [review]

**Pull request merged**